### PR TITLE
Upgrade to ol5

### PR DESCRIPTION
**Pull request by bartvde**
Now ready for review

### CODE REVIEW

File: src/components/map.js [hunk 1]
```diff
@@ -882,7 +871,9 @@ export class Map extends React.Component {
             if (!styleCache[layer.id]) {
               const sprite = new SpriteStyle(spriteOptions[layer.id]);
               styleCache[layer.id] = new Style({image: sprite});
-              this.map.on('postcompose', onPostCompose, sprite);
+              this.map.on('postcompose', (e) => {
```
Comment (boundlessgeoadmin) [inline src/components/map.js]: MAJOR Define this function outside of a loop. [rule](https://sq.boundlessgeo.io/coding_rules#rule_key=javascript%3AFunctionDefinitionInsideLoop)

File: src/components/map.js [hunk 2]
```diff
@@ -896,7 +887,9 @@ export class Map extends React.Component {
               options.rotation = rotation;
               const sprite = new SpriteStyle(options);
               const style = new Style({image: sprite});
-              this.map.on('postcompose', onPostCompose, sprite);
+              this.map.on('postcompose', (e) => {
```
Comment (boundlessgeoadmin) [inline src/components/map.js]: MAJOR Define this function outside of a loop. [rule](https://sq.boundlessgeo.io/coding_rules#rule_key=javascript%3AFunctionDefinitionInsideLoop)

**Comment by boundlessgeoadmin** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/components/map.js [hunk 3]
```diff
@@ -896,7 +887,9 @@ export class Map extends React.Component {
...
+                sprite.update(e);
+              });
               styleCache[layer.id][rotation] = style;
```
Comment (boundlessgeoadmin) [inline src/components/map.js]: MAJOR Define this function outside of a loop. [rule](https://sq.boundlessgeo.io/coding_rules#rule_key=javascript%3AFunctionDefinitionInsideLoop)

**Comment by boundlessgeoadmin** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [map.js#L909](https://github.com/boundlessgeo/sdk/blob/bef805b65ffdb36a7aacf47553cf1f967654be0b/src/components/map.js#L909): Define this function outside of a loop. [rule](https://sq.boundlessgeo.io/coding_rules#rule_key=javascript%3AFunctionDefinitionInsideLoop)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: src/components/map.js [hunk 4]
```diff
@@ -938,7 +927,9 @@ export class Map extends React.Component {
             if (!styleCache[layer.id]) {
               const sprite = new SpriteStyle(spriteOptions[layer.id]);
               styleCache[layer.id] = new Style({image: sprite});
-              this.map.on('postcompose', onPostCompose, sprite);
+              this.map.on('postcompose', (e) => {
```
Comment (boundlessgeoadmin) [inline src/components/map.js]: MAJOR Define this function outside of a loop. [rule](https://sq.boundlessgeo.io/coding_rules#rule_key=javascript%3AFunctionDefinitionInsideLoop)

**Comment by bartvde** [comment]
MVT Tile example's styling is broken ```index.js:145 Uncaught (in promise) Error: Sprites cannot be loaded from https://api.mapbox.com/styles/v1/mapbox/bright-v8/sprite.json?access_token=pk.eyJ1IjoiYmFydHZkZSIsImEiOiJjaXNxNTNwbWcwMDF3MnpwYnc4dTh0dWluIn0.CDTDktSJ68Lffwmnt7BpFA
    at index.js:145```

**Comment by bartvde** [comment]
ah that particular error seems to stem from the legend instead

**Comment by bartvde** [comment]
ah the actual error in the map is: ```"Cannot use 'in' operator to search for 'type' in undefined"```

**Comment by bartvde** [comment]
ah this seems to be caused by the not find refs again

**Comment by bartvde** [comment]
I wonder if this is caused by our code in sdk: ```    // remove the reference
    layer_def.ref = undefined;
```

**Comment by bartvde** [comment]
yes that is actually it, deleting the prop instead of setting it undefined doesn't bork mapbox

**Comment by bartvde** [comment]
Fixed: 
selection_582

Finally

**Comment by ahocevar** [comment]
I released `ol-mapbox-style` v3.0.1 last night.

**Pull request merged**